testtools/files.go

```diff
@@ -129,3 +139,232 @@ func CheckFiles(t *testing.T, dir string, files []FileSpec) {
+			trim := filepath.Join(args.TestDataPath, args.Name) + "/"
+			shortPath := strings.TrimPrefix(f.ShortPath, trim)
+
+			info, err := os.Stat(path)
+			if err != nil {
+				t.Fatalf("os.Stat(%q) error: %v", path, err)
+			}
+
+			if info.IsDir() {
+				continue
+			}
+
+			content, err := ioutil.ReadFile(path)
+			if err != nil {
+				t.Errorf("ioutil.ReadFile(%q) error: %v", path, err)
+			}
+
+			if filepath.Base(shortPath) == "test.yaml" {
+				if config != nil {
+					t.Fatal("only 1 test.yaml is supported")
+				}
+				config = new(testYAML)
+				if err := yaml.Unmarshal(content, config); err != nil {
+					t.Fatal(err)
+				}
+			}
+
+			if strings.HasSuffix(shortPath, args.BuildInSuffix) {
+				inputs = append(inputs, FileSpec{
+					Path:    filepath.Join(args.Name, strings.TrimSuffix(shortPath, args.BuildInSuffix)+".bazel"),
+					Content: string(content),
+				})
+			} else if strings.HasSuffix(shortPath, args.BuildOutSuffix) {
+				goldens = append(goldens, FileSpec{
+					Path:    filepath.Join(args.Name, strings.TrimSuffix(shortPath, args.BuildOutSuffix)+".bazel"),
+					Content: string(content),
+				})
+			} else {
+				inputs = append(inputs, FileSpec{
+					Path:    filepath.Join(args.Name, shortPath),
+					Content: string(content),
+				})
+				goldens = append(goldens, FileSpec{
+					Path:    filepath.Join(args.Name, shortPath),
+					Content: string(content),
+				})
+			}
+		}
+
+		testdataDir, cleanup := CreateFiles(t, inputs)
+		defer cleanup()
+		defer func() {
+			if t.Failed() {
+				shouldUpdate := os.Getenv("UPDATE_SNAPSHOTS") != ""
+				buildWorkspaceDirectory := os.Getenv("BUILD_WORKSPACE_DIRECTORY")
+				if buildWorkspaceDirectory == "" {
+					// Default to ~/<workspace>.
+					homeDir, err := os.UserHomeDir()
+					if err != nil {
+						t.Fatalf("Could not get user's home directory. Error: %v\n", err)
+					}
+					testWorkspace, err := bazel.TestWorkspace()
+					if err != nil {
+						t.Fatalf("Could not get the test workspace. Error; %v\n", err)
+					}
+					buildWorkspaceDirectory = path.Join(homeDir, testWorkspace)
+				}
+				filepath.Walk(testdataDir, func(walkedPath string, info os.FileInfo, err error) error {
+					if err != nil {
+						return err
+					}
+					relativePath := strings.TrimPrefix(walkedPath, testdataDir)
+					if shouldUpdate {
+						if path.Base(walkedPath) == "BUILD.bazel" {
+							destFile := strings.TrimSuffix(path.Join(buildWorkspaceDirectory, args.TestDataPath+relativePath), ".bazel") + ".out"
+
+							err := copyFile(walkedPath, destFile)
+							if err != nil {
+								t.Fatalf("Failed to copy file %v to %v. Error: %v\n", walkedPath, destFile, err)
+							}
+						}
+					}
+					t.Logf("%q exists in %v", relativePath, testdataDir)
+					return nil
+				})
+				if !shouldUpdate {
+					t.Logf(`
+=====================================================================================
+
+Run UPDATE_SNAPSHOTS=true bazel run //path/to/this/test to update BUILD.out files.
```

Comment:
I believe you can use `os.Getenv("TEST_TARGET")`[0][1] to get the actual target inside the test, so this would become

```
t.Logf(`
=====================================================================================
Run UPDATE_SNAPSHOTS=true bazel run %s to update BUILD.out files.
=====================================================================================
`, os.Getenv("TEST_TARGET"))
```

[0] https://pkg.go.dev/os#Getenv
[1] https://docs.bazel.build/versions/main/test-encyclopedia.html#initial-conditions

Reply:
Oh nice!